lib/datastore/dataset.js

```diff
@@ -132,24 +132,27 @@ nodeutil.inherits(Dataset, DatastoreRequest);
  *
  * @private
  *
- * @param {object} options - Configuration object.
- * @param {string=} options.apiEndpoint - Custom API endpoint.
+ * @param {string} customApiEndpoint - Custom API endpoint.
  */
-Dataset.determineApiEndpoint_ = function(options) {
-  var apiEndpoint = 'https://www.googleapis.com';
+Dataset.prototype.determineApiEndpoint_ = function(customApiEndpoint) {
```

Comment:
Wouldn't the side-effect style here make this method harder to unit test? Just a thought (we changed from return "the endpoint" to this.apiEndpoint = "the endpoint";)